### PR TITLE
tryte char conversion improvements

### PR DESCRIPTION
**Pull request by mistersandman**
replace some (expensive) lookups in tryte character conversions with index computations

### CODE REVIEW

File: source/types/trinary.cpp [hunk 1]
```diff
@@ -39,9 +39,22 @@ static std::vector<std::vector<int8_t>> trytesTrits = {
...
+tryteIndex(const char& tryte) {
+  if (tryte == '9') {
+    return 0;
+  }
+  // explicitly excluding '@' here, since by the calculation
+  // below this would map to 0, but 0 is the index for '9'
+  if (tryte == '@') {
+    return -1;
+  }
+  return tryte - 'A' + 1;
```
Comment (Cylix) [inline source/types/trinary.cpp]: I would prefer something consistent: return >= 0 for valid tryte
return -1 for non valid tryte.
The check for @ is quite confusing IMO, I'd prefer something like if >= 'A' && <= 'Z' then - 'A' + 1, otheriwse -1

File: source/types/trinary.cpp [hunk 2]
```diff
@@ -39,9 +39,22 @@ static std::vector<std::vector<int8_t>> trytesTrits = {
...
+}
+
 bool
 isValidTryte(const char& tryte) {
-  return TryteAlphabet.find(tryte) != Types::Trytes::npos;
+  return tryteIndex(tryte) < 27;
```
Comment (Cylix) [inline source/types/trinary.cpp]: Then, for this one, would be >= 0, more readable.

File: source/types/trinary.cpp (hunk 1 above)
Comment (mistersandman) [reply source/types/trinary.cpp]: Thanks for the review. I adjusted it according to your suggestions and I agree, its much better now!

File: source/types/trinary.cpp [hunk 3]
```diff
@@ -39,9 +39,20 @@ static std::vector<std::vector<int8_t>> trytesTrits = {
   { 0, -1, 0 }, { 1, -1, 0 }, { -1, 0, 0 }
 };
 
+int8_t
+tryteIndex(const char& tryte) {
+  if (tryte == '9') {
+    return 0;
+  }
+  if ('A' <= tryte && tryte <= 'Z') {
```
Comment (thibault-martinez) [inline source/types/trinary.cpp]: Did you mean `if ('A' >= tryte && tryte <= 'Z')` ?

**Pull request merged**